### PR TITLE
Master fork

### PR DESCRIPTION
**Pull request by p0psicles**
@xaksis I'm creating the PR to flesh out what I'd like to contribute to vgt and what not.
Over the course of last year, i've added a number of features and fixes, to tailer the lib to our project.
Some of the bugs I encountered I already submitted.

We can use this PR, to see what can be used from it.

### CODE REVIEW

File: src/components/Table.vue [hunk 1]
```diff
@@ -36,24 +36,28 @@
...
-        :search-enabled="searchEnabled && externalSearchQuery == null"
-        :global-search-placeholder="searchPlaceholder"
-      >
-        <template slot="internal-table-actions">
-          <slot name="table-actions">
-          </slot>
-        </template>
-      </vgt-global-search>
+      <div slot="internal-table-actions">
+        <slot name="table-actions-header"></slot>
```
Comment (p0psicles) [inline src/components/Table.vue:40]: Add the ability to use a slot table-actions-header

File: src/components/Table.vue [hunk 2]
```diff
@@ -36,24 +36,28 @@
...
+        </slot>
+      </div>
+      
       <div
         v-if="selectedRowCount && !disableSelectInfo"
         class="vgt-selection-info-row clearfix"
         :class="selectionInfoClass"
       >
-        {{selectionInfo}}
+        <span>{{selectionInfo}}</span>
```
Comment (p0psicles) [inline src/components/Table.vue:60]: Add span, for easy styling

File: src/components/Table.vue [hunk 3]
```diff
@@ -288,6 +296,13 @@
                 </slot>
               </template>
             </vgt-header-row>
+
+            <slot
+              name="table-footer-row"
```
Comment (p0psicles) [inline src/components/Table.vue:301]: I need to have a table-header-row `and` a table-footer-row slot.

File: src/components/Table.vue [hunk 4]
```diff
@@ -437,6 +454,14 @@ export default {
         };
       },
     },
+
+    columnFilterOptions: {
```
Comment (p0psicles) [inline src/components/Table.vue:458]: Don't know why this is here.
Maybe a rebase error/leftover

File: src/components/Table.vue [hunk 5]
```diff
@@ -1438,6 +1474,9 @@ export default {
       });
       return originalRows;
     },
+    toggleFilteredColumn(value) {
```
Comment (p0psicles) [inline src/components/Table.vue:1477]: Part of the columnFilter

File: src/components/VgtFilterRow.vue [hunk 6]
```diff
@@ -206,6 +206,10 @@ export default {
       //* lets emit event once all filters are set
       this.$emit('filter-changed', this.columnFilters);
     },
+    fieldKey(field) {
```
Comment (p0psicles) [inline src/components/VgtFilterRow.vue:209]: SHould be removed, already merged through other PR.

File: dev/grouped-table.vue [hunk 7]
```diff
@@ -24,19 +25,24 @@
...
-      <span v-else>
-        {{props.formattedRow[props.column.field]}}
-      </span>
-    </template> -->
+    <template slot="table-footer-row" slot-scope="{headerRow}">
+      <tr slot="table-footer-row">
+        <td colspan="9999">total: {{ sumCount(headerRow) }}</td>
+      </tr>
+    </template>
+    <template slot="table-actions">
```
Comment (p0psicles) [inline dev/grouped-table.vue:42]: I don't think I use this. Should be removed.

**Comment by p0psicles** [comment]
Would greatly appreciate to get some feedback on the comments. Would help me offer some smaller PR's

**Comment by p0psicles** [comment]
@xaksis would you rather have my PR them one by one? Or do you see some features that should go first?

**Comment by xaksis** [comment]
Hey @p0psicles, you'd actually help me out immensely by breaking them up so we can reject or accept them individually. It is tough to see the scope of each change this way. I think some of these changes have already been implemented. 
Would be great to break them up and submit each as a PR methinks.

**Pull request closed without merging**